spec/features/users/sign_in_spec.rb

```diff
@@ -92,10 +92,22 @@
   end
 
   context 'signed out' do
+    it 'keeps the user on the current page after session times out', js: true do
+      allow(Devise).to receive(:timeout_in).and_return(0)
+
+      visit new_user_registration_path
+      fill_in 'Email', with: 'teststuff'
+
+      find_link(t('forms.buttons.continue')).trigger('click')
+
+      expect(find_field('Email').value).to be_blank
+      expect(current_path).to eq new_user_registration_path
+    end
+
```

Comment:
Combing the tests sounds good!

You removed the `expect` call in between the `find_link` and `expect(current_path)`; unfortunately we need something there to tell Capybara to wait for the redirect. I put the intermediate `expect` to avoid having a `sleep` call.

I verified that this was necessary be replacing the `request.original_url` with `.` as the `url` argument to the `link_to` tag in the expired partial. Even with the original code, the test still passes; obviously we know that is a false positive.

If there is a clearer way to simulate a wait I’m all for it.